File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.cpp
class Solution {
public:
    int countKDifference(vector<int>& nums, int k) {
        vector<int> ch(101);
        for (int x: nums) {
            ch[x] += 1;
        }
        int res = 0;
        for (int x: nums) {
            if (x + k < 101) {
                res += ch[x+k];
            }
        }
        return res;
    }
};

File: 2006-count-number-of-pairs-with-absolute-difference-k/2006-count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        HashMap<Integer, Integer> f = new HashMap<>();
        for (int x: nums) {
            f.put(x, f.getOrDefault(x,0) + 1);
        }
        int res = 0;
        for (int x: nums) {
            if (f.containsKey(x+k)) {
                res += f.get(x+k);
            }
        }
        return res;
    }
}